components/datetime/src/timezone.rs

```diff
@@ -0,0 +1,331 @@
+    pub(super) fn long_specific_non_location_format(
+        &self,
+        time_zone: &impl TimeZoneInput,
+    ) -> Option<Cow<str>> {
+        self.mz_specific_long
+            .as_ref()
+            .and_then(|metazones| time_zone.metazone_id().and_then(|mz| metazones.get(mz)))
+            .and_then(|specific_names| {
+                time_zone
+                    .time_variant()
+                    .and_then(|variant| specific_names.get(variant))
+            })
+            .map(Clone::clone)
+    }
+
+    /// Returns the time zone in localized GMT format according to the CLDR localized hour format.
+    /// This goes explicitly against the UTS-35 spec, which specifies long or short localized
+    /// GMT formats regardless of locale.
+    ///
+    /// You can see more information about our decision to resolve this conflict here:
+    /// https://docs.google.com/document/d/16GAqaDRS6hzL8jNYjus5MglSevGBflISM-BrIS7bd4A/edit?usp=sharing
+    pub(super) fn localized_gmt_format(&self, time_zone: &impl TimeZoneInput) -> Cow<str> {
+        let gmt_offset = time_zone.gmt_offset();
+        if gmt_offset.is_zero() {
+            self.zone_formats.gmt_zero_format.clone()
+        } else {
+            Cow::Owned(
+                self.zone_formats
+                    .gmt_format
+                    .replace(
+                        "{0}",
+                        if gmt_offset.is_positive() {
+                            &self.zone_formats.hour_format.0
+                        } else {
+                            &self.zone_formats.hour_format.1
+                        },
+                    )
+                    // support all combos of "(HH|H):mm" by replacing longest patterns first.
+                    .replace("HH", &gmt_offset.format_hours(ZeroPadding::On))
+                    .replace("mm", &gmt_offset.format_minutes())
+                    .replace("H", &gmt_offset.format_hours(ZeroPadding::Off)),
+            )
+        }
+    }
+
+    /// Returns the exemplar city associated with this time zone.
+    pub(super) fn exemplar_city(&self, time_zone: &impl TimeZoneInput) -> Option<Cow<str>> {
+        self.exemplar_cities
+            .as_ref()
+            .and_then(|cities| time_zone.time_zone_id().and_then(|id| cities.get(id)))
+            .map(Clone::clone)
+    }
+
+    /// Returns the unknown city. This is used as a fallback in case the time-zone's exemplar city
+    /// had no localized form in the current locale.
+    ///
+    /// This function is only invoked as a fallback, which means that we can guarantee that the
+    /// exemplar-city data is present, and all locales should have an entry for "Etc/Unknown".
+    pub(super) fn unknown_city(&self) -> Cow<str> {
+        self.exemplar_cities
+            .as_ref()
+            .expect("Exemplar City data was not present.")
+            .get("Etc/Unknown")
+            .map(Clone::clone)
+            .expect("Etc/Unknown is not contained in exemplar city data.")
```

Comment:
This is a bit strong of an assertion, because it can be triggered by bad data.  I can feed bad JSON to a program and make the program crash.

I suggest defaulting to some fixed static string here.  Consider pulling whatever they use in the und locale.

Reply:
Hmm, I agree with you. We should probably have some static fallback here. 

There doesn't seem to be an `und` locale in [cldr-dates-full/main/](https://github.com/unicode-org/cldr-json/tree/master/cldr-json/cldr-dates-full/main)

Am I interpreting this incorrectly?
Otherwise, do you have any other ideas?

Reply:
CLDR uses "root" instead of "und".